tests/Unit/Utilities/Test_TupleSlice.cpp

```diff
@@ -9,8 +9,21 @@
 #include "Utilities/TupleSlice.hpp"
 
 SPECTRE_TEST_CASE("Unit.Utilities.TupleSlice", "[Utilities][Unit]") {
-  const std::tuple<int, float, double, std::string> tuple{1, 2., 3., ""};
-  CHECK(tuple_slice<1, 3>(tuple) == std::tuple<float, double>{2., 3.});
-  CHECK(tuple_head<2>(tuple) == std::tuple<int, float>{1, 2.});
-  CHECK(tuple_tail<2>(tuple) == std::tuple<double, std::string>{3., ""});
+  {
+    std::tuple<int, float, double, std::string> tuple{1, 2., 3., ""};
```

Comment:
Sorry for not thinking of this during my last review: can you test 1 and 0 element tuples? I'm not sure what they will do, so we should check that. Can you also test having `Size` in `head` and `tail` be larger than the size of the tuple? I think you'll find undesirable behavior in those cases and may want to add a `static_assert` :)

Reply:
Added the tests. The invalid `Size` and slice ranges trigger standard errors like this:

```
error: static_assert failed due to requirement '0UL < sizeof...(_Types)' "tuple_element index out of range"
```

I suppose that's as good as it gets and we don't need to add extra asserts?

Reply:
Are you sure that error isn't STL implementation dependent?

Reply:
no I'm not sure, can I check that? Would you suggest to add another assert then?

Reply:
with AppleClang it triggers here: `/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/__tuple:283:5`

Reply:
test in the container with libc++ and stdlibc++, and check what the standard says (cppreference should be good enough)

Reply:
[`std::get`](https://en.cppreference.com/w/cpp/utility/tuple/get) "Extracts the Ith element from the tuple. I must be an integer value in [0, sizeof...(Types))."

Reply:
Just add static asserts... Quoting cppreference doesn't help me, I can do that myself. My main concern is if this is used in generic code you'll end up creating a giant index sequence if something goes wrong, which just segfaults the compiler. I think you want to require the tuple have less than 1000 elements, and that the bounds are check

Reply:
Sure I‘ll add asserts, I just wasn’t sure before if that would be useful or add unnecessary code. In general, are static asserts cheap enough on compile time to add them liberally?